src/rgw/motr/gc/gc.cc

```diff
@@ -47,22 +47,42 @@ void *MotrGC::GCWorker::entry() {
         << worker_id << " Working on GC Queue: " << iname << dendl;
       // time based while loop
       do {
-        // fetch the next entry from index "iname"
-        // check if the object is ready for deletion
-        // for the motr object
-
-        // delete that object
-        // rc = dequeue();
-
-        // remove the entry from iname
+        bufferlist bl;
+        std::vector<std::string>keys(motr_gc->max_count + 1);
+        std::vector<bufferlist>vals(motr_gc->max_count + 1);
+        uint32_t rgw_gc_obj_min_wait = cct->_conf->rgw_gc_obj_min_wait;
+        rc = motr_gc->store->next_query_by_name(iname, keys, vals, "0_");
+        if (rc < 0) {
+          // In case of failure, worker will keep retrying till end_time
+          ldpp_dout(dpp, 0) <<__func__<<": ERROR: NEXT query failed. rc=" << rc << dendl;
+        }
 
+        // fetch entries as per defined in rgw_gc_max_trim_chunk from index iname
+        for (uint32_t j = 0; j < motr_gc->max_count && !keys[j].empty(); j++) {
+          bufferlist::const_iterator blitr = vals[j].cbegin();
+          motr_gc_obj_info ginfo;
+          ginfo.decode(blitr);
+          // check if the object is ready for deletion
+          if(ginfo.time + rgw_gc_obj_min_wait < std::time(nullptr)) {
+            // delete motr object
+            rc = motr_gc->delete_motr_obj_from_gc(ginfo);
+            if (rc < 0) {
+              ldpp_dout(dpp, 0) << "ERROR: Motr obj deletion failed for "
+                                    << ginfo.tag << " with rc: " << rc << dendl;
+              continue; // should continue deletion for next objects
+            }
+            // delete entry from GC queue
+            rc = motr_gc->dequeue(iname, ginfo);
+          }
+        }
         // Exit the loop if required work is complete
         processed_count++;
```

Comment:
we don't need this processed count now I think, since we are fetching entries equal to max_chunk_size.

Reply:
We actually need to decide what should `processed_count` represent.
Let's say `rgw_gc_max_trim_chunk` is 30, so we picked 30 entries out of which 10 are expired.
So `processed_count` will be 30 since 30 objects were queried or should be 10 since only 10 were deleted.
@sachinpunadikar @sumedhak27